clippy_lints/src/trait_bounds.rs

```diff
@@ -167,6 +167,63 @@ impl<'tcx> LateLintPass<'tcx> for TraitBounds {
             }
         }
     }
+
+    fn check_ty(&mut self, cx: &LateContext<'tcx>, ty: &'tcx Ty<'tcx>) {
+        let TyKind::Ref(
+            ..,
+            MutTy {
+                ty: Ty {
+                    kind: TyKind::TraitObject(
+                        bounds,
+                        ..
+                    ),
+                    ..
+                },
+                ..
+            }
+        ) = ty.kind else { return; };
+
+        if bounds.len() < 2 {
+            return;
+        }
+
+        let mut bounds_span = bounds[0].span;
+
+        for bound in bounds.iter().skip(1) {
+            bounds_span = bounds_span.to(bound.span);
+        }
+
+        let mut seen_def_ids = FxHashSet::default();
+        let mut fixed_traits = Vec::new();
+
+        for bound in bounds.iter() {
+            let Some(def_id) = bound.trait_ref.trait_def_id() else { continue; };
+
+            let new_trait = seen_def_ids.insert(def_id);
+
+            if new_trait {
+                fixed_traits.push(bound);
+            }
+        }
+
+        let fixed_trait_snippet = fixed_traits
+            .iter()
+            .filter_map(|b| snippet_opt(cx, b.span))
+            .collect::<Vec<_>>()
+            .join(" + ");
```

Comment:
You can delay constructing this string until suggestion can actually be applied (i.e. move into next if)